### ISSUE TITLE
ERROR: Sorry, Windows does not support runas functionality

### ISSUE THREAD

**Issue by LiangAllen** [saltstack/salt#30318]
`salt-api cmd.run windows minion` ,return  `ERROR: Sorry, Windows does not support runas functionality`
```
params = {'client': 'local', 'fun': 'cmd.run', 'tgt': "windows.minion",'arg':'...'}
```
How to solve?

**Comment by jfindlay** [comment]
@LiangAllen, thanks for the report.  What version of salt are you using?  Will you also post your salt-api configs?

**Comment by LiangAllen** [comment]
@jfindlay thanks  for your reply.

version:
salt 2015.8.3
salt-api 0.8.4.1

salt-api config: Only configure the user name and password

POST / HTTP/1.1
Host: localhost:8000
Accept: application/json
X-Auth-Token: d40d1e1e
Content-Type: application/x-www-form-urlencoded

fun=cmd.run&client=local&tgt=*&arg=C:/Powershell/a.ps1

return "ERROR: Sorry, Windows does not support runas functionality"

I want to execute powershell script on windows minion

**Comment by jfindlay** [comment]
@LiangAllen, thanks for the extra information.  I wonder if salt-api is supplying a runas argument to cmd.run because you've specified a username, whereas windows doesn't support runas (yet).

**Comment by whiteinge** [comment]
@jfindlay No. salt-api does not inject kwargs. What the user supplies is sent to the minion verbatim. Username is the eauth username and used by the Salt Master only.

@LiangAllen your salt-api version number should *match* the Salt version number. It was once a separate project but was merged into the main Salt repo in the 2014.7 release. Uninstall salt-api and reinstall Salt then check the version numbers for both programs again.

If you still run into trouble after those steps, please paste the _entire_ HTTP request that you are sending to salt-api (either JSON or urlencoded). As mentioned, the Salt Windows daemon does not yet support switching users so that argument to cmd.run is not supported on WIndows.

**Comment by cachedout** [comment]
Closed for lack of response here.
